syntax = "proto2";


// message UFLDParam {
//   optional ModelParam model_param = 1;
//   optional NetworkParam net_param = 2;
//   optional PostprocessorParam post_param = 3;
// }

// message PostprocessorParam {
//   optional uint32 lane_map_width = 1 [default = 640];
//   optional uint32 lane_map_height = 2 [default = 480];
//   optional uint32 roi_height = 3 [default = 768];
//   optional uint32 roi_start = 4 [default = 312];
//   optional uint32 roi_width = 5 [default = 1920];
// }

message NetConfig {
  repeated string input_names = 1;
  repeated string output_names = 2;
}

message BEVConfig {
  optional string engine_file = 1 [default = "resnet18.engine"];
  optional uint32 src_height = 2 [default = 0];
  optional uint32 src_width = 3 [default = 0];
  optional uint32 resize_height = 4 [default = 256];
  optional uint32 resize_width = 5 [default = 704];
  optional uint32 input_offset_x = 6 [default = 0];
  optional uint32 input_offset_y = 7 [default = 0];
  optional uint32 input_channel = 8 [default = 3];

  optional float mean_b = 9 [default = 95];
  optional float mean_g = 10 [default = 99];
  optional float mean_r = 11 [default = 96];
  optional float std_b = 12 [default = 96];
  optional float std_g = 13 [default = 96];
  optional float std_r = 14 [default = 96];
  optional bool is_bgr = 15 [default = true];

  optional float confidence_threshold = 16 [default = 0.5];
  optional int32  gpu_id = 17 [default = 0];

  optional NetConfig net_config = 18;
}

message BEVDet{
  optional BEVConfig part1_config = 1;
  optional BEVConfig part2_config = 2;  
}